openedx/core/djangoapps/course_date_signals/tests.py

```diff
@@ -114,5 +116,267 @@ def test_sequence_with_graded_and_ungraded_assignments(self):
+
+                chapter2 = ItemFactory.create(category='chapter', parent=course4)
+                sequential2 = ItemFactory.create(category='sequential', parent=chapter2, graded=True)
+                vertical2 = ItemFactory.create(category='vertical', parent=sequential2)
+                ItemFactory.create(category='problem', parent=vertical2)
+            self.courses.append(course4)
+
+            # course 4: a section with a subsection that has due_num_weeks and multiple sections without due_num_weeks that has graded content
+            course5 = CourseFactory.create(self_paced=True)
+            with self.store.bulk_operations(course5.id):
+                chapter1 = ItemFactory.create(category='chapter', parent=course5)
+                sequential1 = ItemFactory.create(category='sequential', parent=chapter1, due_num_weeks=2)
+                vertical1 = ItemFactory.create(category='vertical', parent=sequential1)
+                ItemFactory.create(category='problem', parent=vertical1)
+
+                chapter2 = ItemFactory.create(category='chapter', parent=course5)
+                sequential2 = ItemFactory.create(category='sequential', parent=chapter2, graded=True)
+                vertical2 = ItemFactory.create(category='vertical', parent=sequential2)
+                ItemFactory.create(category='problem', parent=vertical2)
+
+                chapter3 = ItemFactory.create(category='chapter', parent=course5)
+                sequential3 = ItemFactory.create(category='sequential', parent=chapter3, graded=True)
+                vertical3 = ItemFactory.create(category='vertical', parent=sequential3)
+                ItemFactory.create(category='problem', parent=vertical3)
+
+                chapter4 = ItemFactory.create(category='chapter', parent=course5)
+                sequential4 = ItemFactory.create(category='sequential', parent=chapter4, graded=True)
+                vertical4 = ItemFactory.create(category='vertical', parent=sequential4)
+                ItemFactory.create(category='problem', parent=vertical4)
+            self.courses.append(course5)
+
+    @override_waffle_flag(CUSTOM_PLS, active=True)
+    def test_extract_dates_from_course_inheritance(self):
+        """
+        extract_dates_from_course should return a list of (block item location, field metadata dictionary)
+        and the blocks should inherit the dates from those above in the hiearchy
+        (ex. If a subsection is assigned a due date, its children should also have the same due date)
+        """
+        # course 1: With due_num_weeks but without any units. The subsection should have a set due date,
+        # while section should have the minimum number of weeks to complete as the due date (4 weeks).
+        course = self.courses[0]
+        chapter = course.get_children()[0]
+        sequential = chapter.get_children()[0]
+        expected_dates = [
+            (course.location, {}),
+            (chapter.location, timedelta(days=28)),
```

Comment:
Also this feels weird and we should talk about it in the future as we iterate on this feature. Like my first impression is that it’s odd the section and subsection have different due dates. My first guess of a better solution would be the section due date should be the max of all of the subsection due dates inside of it, but again, we should chat and figure out what the best solution should be for this